lib/library.rb

```diff
@@ -0,0 +1,55 @@
+class Library
+  attr_reader :name,
+              :books,
+              :authors,
+              :checked_out_books
+
+  def initialize(name)
+    @name = name
+    @books = []
+    @authors = []
+    @checked_out_books = []
+  end
+
+  def add_author(author)
+    @authors << author
+    add_author_books(author)
+  end
+
+  def add_author_books(author)
```

Comment:
What happens though if the author writes another book after they're added? If we calculated the books dynamically based off of the @authors array, then we wouldn't have to worry about that. :)